### PR TITLE
feat: `avm/res/consumption/budget`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.consumption.budget](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.consumption.budget.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: avm/res/consumption/budget/main.bicep [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+@description('Conditional. The list of email addresses to send the budget notification to when the thresholds are exceeded. Required if neither `contactRoles` nor `actionGroups` was provided.')
+param contactEmails array = []
+
+@description('Conditional. The list of contact roles to send the budget notification to when the thresholds are exceeded. Required if neither `contactEmails` nor `actionGroups` was provided.')
+param contactRoles array = []
+
+@description('Conditional. List of action group resource IDs that will receive the alert. Required if neither `contactEmails` nor `contactEmails` was provided.')
+param actionGroups array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```
Comment (AlexanderSehr) [inline avm/res/consumption/budget/main.bicep:56]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: avm/res/consumption/budget/main.bicep [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+param location string = deployment().location
+
+var notificationsArray = [for threshold in thresholds: {
+  'Actual_GreaterThan_${threshold}_Percentage': {
+    enabled: true
+    operator: 'GreaterThan'
+    threshold: threshold
+    contactEmails: empty(contactEmails) ? null : array(contactEmails)
+    contactRoles: empty(contactRoles) ? null : array(contactRoles)
+    contactGroups: empty(actionGroups) ? null : array(actionGroups)
```
Comment (AlexanderSehr) [inline avm/res/consumption/budget/main.bicep:69]:
```suggestion
    contactEmails: !empty(contactEmails) ? array(contactEmails) : null
    contactRoles: !empty(contactRoles) ? array(contactRoles) : null
    contactGroups: !empty(actionGroups) ? array(actionGroups) : null
```

File: .github/workflows/avm.res.consumption.budget.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  ###########################
+  #   Initialize pipeline   #
+  ###########################
+  job_initialize_pipeline:
+    runs-on: ubuntu-20.04
```
Comment (eriqua) [inline .github/workflows/avm.res.consumption.budget.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**